HACKING.rst

```diff
@@ -86,18 +86,129 @@ You can also see all the environments by simply running ``tox list``
 Running ``tox run -m format`` and ``tox run -m lint`` before committing code is
 recommended.
 
-Code Style
-==========
-
 Commits
-#######
+-------
+
+Commit messages are based on the `conventional commit`_ style::
+
+  <type>(<optional scope>): <description>
+
+  <optional body>
+
+  <optional footer>
+
+The commit is divided into three sections: a header, body, and footer.
+
+Header
+======
+
+The header is required and consists of three subsections: a type,
+optional scope, and description. The header must be 72 characters or less.
+
+Types
+#####
+
+``build``
+"""""""""
+
+Commits that affect the build of an application or library.
+
+This includes dependency updates, which should use the ``deps`` scope
+(``build(deps):``).
+
+``chore``
+"""""""""
+
+Miscellaneous commits that don't fit into any other type.
+
+Examples include:
+
+* development dependency updates
+* edits to a comment or docstring
+* type changes
+* accommodating a developer-facing deprecation warning
+* many *small* fixes for an existing PR
```

Comment:
I like @syu-w's suggestion of using `chore` for merges. Perhaps we should call out `chore(merge)` both as an example here and as an exception to the priorities I mentioned elsewhere? (Namely, if it's a merge commit, your diff on the branch might have some of everything, but that commit itself should just be the merge and thus be considered a chore)

Reply:
Noted!